Sources/App/routes.swift

```diff
@@ -31,45 +52,109 @@ public func routes(_ router: Router) throws {
         
-        return circle.getOutput(for: webhook.buildNumber, repo: repo, step: "swift test", on: req)
-            .flatMap { (output, pullRequest) -> Future<CreateCommentResponse> in
-                guard
-                    let issueNumberString = pullRequest.url.absoluteString.split(separator: "/").last,
-                    let issueNumber = Int(issueNumberString)
-                    else {
-                        throw Abort(.notFound)
-                    }
-                
-                let performanceTestParser = ParsePerformanceTestResultsInteractor()
-                guard let testResults = try? performanceTestParser.execute(output: output.message) else {
-                    return github.postComment(
-                        repo: repo,
-                        issue: issueNumber,
-                        body: "Performance tests failed in an unexpected way",
-                        on: req
-                    )
-                }
-                
-                let tableGenerator = GithubTableGenerator(
-                    columns: "Name", "Expected", "Actual", "Change",
-                    rows: testResults
-                )
-                
-                return github.postComment(
-                    repo: repo,
-                    issue: issueNumber,
-                    body: tableGenerator.table,
-                    on: req
-                )
-            }.transform(to: .ok)
+        return circle
+            .getOutput(for: webhook.buildNumber, repo: repo, step: "swift test", on: req)
+            .flatMap { (output, build) -> Future<HTTPStatus> in
+                return try req
+                    .keyedCache(for: .sqlite)
+                    .get(String(webhook.buildNumber), as: Bool.self)
+                    .flatMap { maybeTracked -> Future<[PerformanceTestResults]> in
+                        let performanceTestParser = ParsePerformanceTestResultsInteractor()
+                        guard let testResults = try? performanceTestParser
+                            .execute(
+                                output: output.message,
+                                date: build.startTime,
+                                repoName: build.repoName,
+                                on: req
+                        ) else {
+                            return req.future(error: ParsingError.parsingFailed)
+                        }
+
+                        // if this test is the result of a merge
+                        if maybeTracked != nil {
+                            return testResults.flatMap { result in
+                                return result.map { $0.create(on: req) }.flatten(on: req)
+                            }.flatMap { results in
+                                print(results)
+                                // Skipping down to the catch map
+                                return req.future(error: SkipError.skip)
```

Comment:
This feels bad to me and I'd like to get a second opinion on if this is bad practice or not. What we're doing here is skipping table generation and commenting on github (the next part of the pipeline) if this CircleCI build is the was triggered due to a merge (eg there's no PR to comment on), but this pattern kinda feels like a GOTO with extra steps. I'd love to hear an alternative if one exists.

Reply:
Is the part that this skips to executed in any flow? If so, maybe extract it into a function and call it?

Reply:
Or make this a promise thing, and fulfill the promise. That might work

Reply:
Yeah I think @MrLotU is right. You should just put the rest of the function into a method call that you only call in the success case (or just nest it in there). Then the case of a merge run would just return `HTTPStatus.ok` immediately.